website/src/documentation/runners/flink.md

```diff
@@ -143,29 +174,115 @@ Use the Beam version and the artifact id from the above table. For example:
+### Executing a Beam pipeline on a Flink Cluster
 
-```
+<span class="language-java">
+For executing a pipeline on a Flink cluster you need to package your program
+along with all dependencies in a so-called fat jar. How you do this depends on
+your build system but if you follow along the [Beam Quickstart]({{ site.baseurl
+}}/get-started/quickstart/) this is the command that you have to run:
+</span>
+
+```java
 $ mvn package -Pflink-runner
 ```
-The Beam Quickstart Maven project is setup to use the Maven Shade plugin to create a fat jar and the `-Pflink-runner` argument makes sure to include the dependency on the Flink Runner.
+<span class="language-java">Look for the output JAR of this command in the
+install apache_beam``target` folder.
+<span>
+
+<span class="language-java">
+The Beam Quickstart Maven project is setup to use the Maven Shade plugin to
+create a fat jar and the `-Pflink-runner` argument makes sure to include the
+dependency on the Flink Runner.
+</span>
 
-For actually running the pipeline you would use this command
+<span class="language-java">
+For running the pipeline the easiest option is to use the `flink` command which
+is part of Flink:
+</span>
+
+```java
+$ bin/flink -c org.apache.beam.examples.WordCount /path/to/your.jar
+--runner=FlinkRunner --other-parameters
 ```
-$ mvn exec:java -Dexec.mainClass=org.apache.beam.examples.WordCount \
+
+<span class="language-java">
+Alternatively you can also use Maven's exec command. For example, to execute the
+WordCount example:
+</span>
+
+```java
+mvn exec:java -Dexec.mainClass=org.apache.beam.examples.WordCount \
     -Pflink-runner \
     -Dexec.args="--runner=FlinkRunner \
       --inputFile=/path/to/pom.xml \
       --output=/path/to/counts \
       --flinkMaster=<flink master url> \
       --filesToStage=target/word-count-beam-bundled-0.1.jar"
 ```
+<!-- Span implictly ended -->
+
+<span class="language-java">
 If you have a Flink `JobManager` running on your local machine you can provide `localhost:8081` for
-`flinkMaster`. Otherwise an embedded Flink cluster will be started for the WordCount job.
+`flinkMaster`. Otherwise an embedded Flink cluster will be started for the job.
+</span>
+
+<span class="language-py">
+As of now you will need a copy of Apache Beam's source code. You can
+download it on the [Downloads page]({{ site.baseurl
+}}/get-started/downloads/). In the future there will be pre-built Docker images
+available.
+</span>
+
+<span class="language-py">1. *Only required once:* Build the SDK harness container: `./gradlew :beam-sdks-python-container:docker`
+</span>
+
+<span class="language-py">2. Start the Flink portable JobService endpoint: `./gradlew :beam-runners-flink_2.11-job-server:runShadow`
+</span>
+
+<span class="language-py">
+The JobService is the central instance where you submit your Beam pipeline
+to. This is similar to Flink's JobManager. You will still need a JobManager if
```

Comment:
This could potentially lead to confusion. How about "The JobService is the central instance where you submit your Beam pipeline to. The JobService will create a Flink job for the pipeline and can also execute the job in embedded mode. To execute on a Flink cluster, the Beam JobService needs to be provided with the Flink JobManager address to submit the job to."

Reply:
Sounds good.